File: 100 days of leetcode/Day12.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int x = nums.size();

        int i = 0, j = 0;
        while(i < x && j < x){
            while(i < x && nums[i] != 0){
                i++;
            }

            while(j < x && (nums[j] == 0 || j <= i)){
                j++;
            }

            if(i < x && j < x){
                int temp = nums[j];
                nums[j] = nums[i];
                nums[i] = temp;
                i++;
                j++;
            }
        }
    }
};

File: 100 days of leetcode/Day12.py
class Solution(object):
    def moveZeroes(self, nums):
        """
        :type nums: List[int]
        :rtype: None Do not return anything, modify nums in-place instead.
        """
        x = len(nums)

        i = 0
        j = 0
        while(i < x and j < x):
            while(i < x and nums[i] != 0):
                i+=1
            

            while(j < x and (nums[j] == 0 or j <= i)):
                j+=1

            if(i < x and j < x):
                temp = nums[j]
                nums[j] = nums[i]
                nums[i] = temp
                i+=1
                j+=1